kore/src/Kore/Step/Simplification/Not.hs

```diff
@@ -111,17 +112,35 @@ makeEvaluateNot
 makeEvaluateNot Not { notChild } =
     OrPattern.fromPatterns
         [ Pattern.fromTermLike $ makeTermNot term
-        , Conditional
-            { term = mkTop (termLikeSort term)
-            , predicate =
-                makeNotPredicate
-                $ makeAndPredicate predicate
-                $ Predicate.fromSubstitution substitution
-            , substitution = mempty
-            }
+        , Pattern.fromPredicateSorted
+            (termLikeSort term)
+            (makeEvaluatePredicate predicate)
         ]
   where
-    Conditional { term, predicate, substitution } = notChild
+    (term, predicate) = Conditional.splitTerm notChild
+
+makeEvaluatePredicate
+    ::  ( Ord variable
+        , Show variable
+        , SortedVariable variable
+        , Unparse variable
+        )
+    => Predicate variable
+    -> Predicate variable
+makeEvaluatePredicate
+    Conditional
+        { term = ()
+        , predicate
+        , substitution
+        }
+  = Conditional
+        { term = ()
+        , predicate =
+            makeNotPredicate
```

Comment:
Why is there a negation here? I'm asking because this would confuse me just by reading the function name.

Reply:
Added function documentation, is it better now?